### PR TITLE
Custom PRNG: make KeyArray compatible with custom_jvp

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

File: tests/random_test.py [hunk 1]
```diff
@@ -1905,6 +1905,31 @@ def test_make_array_from_single_device_arrays(self):
     result = jax.make_array_from_single_device_arrays(shape, sharding, arrays)
     self.assertArraysEqual(result, keys)
 
+  def test_key_array_custom_prng(self):
```
Comment (froystig) [inline tests/random_test.py:1908]:
```suggestion
  def test_key_array_custom_jvp(self):
```

File: jax/_src/prng.py [hunk 2]
```diff
@@ -317,6 +318,10 @@ def transpose(self, *_, **__) -> PRNGKeyArray: assert False
     'squeeze', 'swapaxes', 'take', 'transpose', 'T'])
 basearray.Array.register(PRNGKeyArrayImpl)
 
+def zeros_like_prngkey(x: PRNGKeyArrayImpl) -> PRNGKeyArrayImpl:
+  return PRNGKeyArrayImpl(x.impl, jnp.zeros_like(x._base_array))
```
Comment (froystig) [inline jax/_src/prng.py:322]: Should we make `jnp.zeros_like` work on key arrays (as discussed elsewhere) and then implement this on top of that directly?

Comment (jakevdp) [reply jax/_src/prng.py:322]: Yeah, that's a good idea – though did we make a final decision on `zeros_like`?

Comment (froystig) [reply jax/_src/prng.py:322]: For both `zeros` and `zeros_like`, the latest thought was "a key array wrapping zeros but considered already-consumed by the reuse checker" (understanding that the latter does not yet exist).

Comment (froystig) [reply jax/_src/prng.py:322]: And we could possibly treat `empty` similarly, if needed.

File: tests/random_test.py (hunk 1 above)
Comment (froystig) [reply tests/random_test.py:1908]: This might also apply to the PR name.

File: jax/_src/prng.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/prng.py:322]: OK, I'll do that first and then rewrite this PR on top of it.

File: tests/random_test.py (hunk 1 above)
Comment (jakevdp) [reply tests/random_test.py]: Done

File: jax/_src/prng.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/prng.py]: Done, see #16033

**Review by froystig — APPROVED** [review]

**Pull request merged**